backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -0,0 +1,366 @@
+import org.ovirt.engine.core.common.businessentities.storage.VolumeFormat;
+import org.ovirt.engine.core.common.businessentities.storage.VolumeType;
+import org.ovirt.engine.core.common.errors.EngineMessage;
+import org.ovirt.engine.core.common.locks.LockingGroup;
+import org.ovirt.engine.core.common.utils.Pair;
+import org.ovirt.engine.core.compat.CommandStatus;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.dao.DiskImageDao;
+import org.ovirt.engine.core.dao.ImageDao;
+import org.ovirt.engine.core.dao.StorageDomainDao;
+import org.ovirt.engine.core.dao.VmDao;
+import org.ovirt.engine.core.utils.transaction.TransactionSupport;
+
+@NonTransactiveCommandAttribute
+public class ConvertDiskCommand<T extends ConvertDiskCommandParameters> extends BaseImagesCommand<T> implements SerialChildExecutingCommand {
+
+    @Inject
+    private DiskImageDao diskImageDao;
+
+    @Inject
+    private ImageDao imageDao;
+
+    @Inject
+    private VmDao vmDao;
+
+    @Inject
+    private ImagesHandler imagesHandler;
+
+    @Inject
+    private StorageDomainDao storageDomainDao;
+
+    @Inject
+    private VdsCommandsHelper vdsCommandsHelper;
+
+    @Inject
+    @Typed(SerialChildCommandsExecutionCallback.class)
+    private Instance<SerialChildCommandsExecutionCallback> callbackProvider;
+
+    public ConvertDiskCommand(T parameters, CommandContext cmdContext) {
+        super(parameters, cmdContext);
+    }
+
+    public ConvertDiskCommand(Guid commandId) {
+        super(commandId);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        if (getDiskImage() == null) {
+            DiskImage diskImage = diskImageDao.getSnapshotById(getLocationInfo().getImageId());
+            setDiskImage(diskImage);
+        }
+    }
+
+    @Override
+    protected boolean validate() {
+        StorageDomain storageDomain = storageDomainDao.get(getDiskImage().getStorageIds().get(0));
+        if (storageDomain == null) {
+            return failValidation(EngineMessage.STORAGE_DOMAIN_DOES_NOT_EXIST);
+        }
+
+        // We do not support chains currently
+        if (diskImageDao.getAllSnapshotsForImageGroup(getDiskImage().getId()).stream().count() > 1) {
+            return failValidation(EngineMessage.ACTION_TYPE_FAILED_DISK_WITH_CHAIN);
+        }
+
+        StorageDomainValidator storageDomainValidator = new StorageDomainValidator(storageDomain);
+        if (!validate(storageDomainValidator.hasSpaceForNewDisk(getDiskImage()))) {
+            return false;
+        }
+
+        if (getDiskImage().getImageStatus() == ImageStatus.LOCKED) {
+            return failValidation(EngineMessage.ACTION_TYPE_FAILED_DISK_IS_LOCKED);
+        }
+
+        // We do not support live stuff currently
+        Map<Boolean, List<VM>> vms = vmDao.getForDisk(getDiskImage().getId(), true);
+
+        boolean runningVms = vms.computeIfAbsent(Boolean.TRUE, b -> Collections.emptyList())
```

Comment:
`Boolean.TRUE, b -> Collections.emptyList()`
just from curiosity - what is that for?

Reply:
prevent NPE if there are no VMs